services/src/projects/postgres_projectdb.rs

```diff
@@ -40,6 +42,96 @@ where
     pub fn new(conn_pool: Pool<PostgresConnectionManager<Tls>>) -> Self {
         Self { conn_pool }
     }
+
+    async fn list_plots(
+        &self,
+        conn: &PooledConnection<'_, PostgresConnectionManager<Tls>>,
+        project_version_id: &ProjectVersionId,
+    ) -> Result<Vec<String>> {
+        let stmt = conn
+            .prepare(
+                "
+                    SELECT name
+                    FROM project_version_plots
+                    WHERE project_version_id = $1;
+                ",
+            )
+            .await?;
+
+        let plot_rows = conn.query(&stmt, &[project_version_id]).await?;
+        let plot_names = plot_rows.iter().map(|row| row.get(0)).collect();
+
+        Ok(plot_names)
+    }
+
+    async fn load_plots(
```

Comment:
validate permission via `PostgresContext::check_user_project_permission`

Reply:
checked in load